src/Arcus.Messaging.Health/Tcp/TcpHealthListenerOptions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using GuardNet;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Arcus.Messaging.Health.Tcp
+{
+    /// <summary>
+    /// Options to control how the <see cref="TcpHealthListener"/> acts as a TCP endpoint for the <see cref="HealthReport"/>.
+    /// </summary>
+    public class TcpHealthListenerOptions
+    {
+        private string _tcpHealthPort = "ARCUS_HEALTH_PORT";
+
+        /// <summary>
+        /// Gets or sets the function that serializes the <see cref="HealthReport"/> to a series of bytes.
+        /// </summary>
+        public IHealthReportSerializer HealthReportSerializer { get; set; }
```

Comment:
Maybe just call it `Serializer`? We're already in the TCP health context.

Reply:
Yeah, but in the lamdba it would look like `options => options` I think it adds more description this way?

Reply:
I presume you are talking about the registration then?

```csharp
services.AddTcpHealthProbes("MyConfigurationKey",
        configureTcpListenerOptions: options =>
        {
            // Configure how the health report should be serialized.
            options.HealthReportSerializer = new MyHealthReportSerializer();
        });
```

```csharp
services.AddTcpHealthProbes("MyConfigurationKey",
        configureTcpListenerOptions: options =>
        {
            // Configure how the health report should be serialized.
            options.Serializer = new MyHealthReportSerializer();
        });
```

I think both are fine, just *personally* think `HealthReport` doesn't have much added value but up to you!

Reply:
I've changed it.

Reply:
Thanks!